src/app.rs

```diff
@@ -148,10 +148,11 @@ impl App<'_> {
         }
     }
 
-    pub fn update_done_state(&mut self) {
-        let selected_task = self.get_selected_task().unwrap();
+    pub fn update_done_state(&mut self) -> Option<()> {
+        let selected_task = self.get_selected_task()?;
```

Comment:
I think you can return early with let-else, this function return type doesn't mean anything so we don't need one